cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/client/AzureImageService.java

```diff
@@ -0,0 +1,179 @@
+        persistenceNotifier.notifyAllocation(buildCloudResource(imageName, imageId, CommonStatus.REQUESTED), ac.getCloudContext());
+    }
+
+    private void updateImageStatus(AuthenticatedContext ac, String imageName, String imageId, CommonStatus commonStatus) {
+        LOGGER.debug("Updating image status to CREATED: {}", imageId);
+        persistenceNotifier.notifyUpdate(buildCloudResource(imageName, imageId, commonStatus), ac.getCloudContext());
+    }
+
+    private String getRegion(AuthenticatedContext ac) {
+        return ac.getCloudContext()
+                .getLocation()
+                .getRegion()
+                .getRegionName();
+    }
+
+    private String getImageId(String resourceGroup, AzureClient client, String imageName) {
+        return azureImageIdProviderService.generateImageId(client.getCurrentSubscription()
+                .subscriptionId(), resourceGroup, imageName);
+    }
+
+    public CloudResource buildCloudResource(String name, String id, CommonStatus status) {
+        return CloudResource.builder()
+                .name(name)
+                .status(status)
+                .persistent(true)
+                .reference(id)
+                .type(ResourceType.AZURE_MANAGED_IMAGE)
+                .build();
+    }
+
+    private boolean isRequested(String imageId) {
+        return getImagesFromDatabase(imageId).isPresent();
+    }
+
+    private Optional<CloudResource> getImagesFromDatabase(String imageId) {
+        return resourcePersistenceRetriever.notifyRetrieve(imageId, CommonStatus.REQUESTED, ResourceType.AZURE_MANAGED_IMAGE);
+    }
+
+    private VirtualMachineCustomImage createCustomImage(String imageName, String resourceGroup, String fromVhdUri, String region, AzureClient client) {
+        return handleAuthException(() -> {
+            Azure azure = client.getAzure();
+            LOGGER.info("check the existence of resource group '{}', creating if it doesn't exist on Azure side", resourceGroup);
+            if (!azure.resourceGroups().contain(resourceGroup)) {
+                LOGGER.info("Creating resource group: {}", resourceGroup);
+                azure.resourceGroups()
+                        .define(resourceGroup)
+                        .withRegion(region)
+                        .create();
+            }
+            LOGGER.debug("Create custom image from '{}' with name '{}' into '{}' resource group (Region: {})",
+                    fromVhdUri, imageName, resourceGroup, region);
+            return measure(() -> azure.virtualMachineCustomImages()
+                    .define(imageName)
+                    .withRegion(region)
+                    .withExistingResourceGroup(resourceGroup)
+                    .withLinuxFromVhd(fromVhdUri, OperatingSystemStateTypes.GENERALIZED)
+                    .create(),
+                    LOGGER, "Custom image has been created under {} ms with name {}", imageName);
+        });
```

Comment:
I would move this code into AzureClient, to have all operations on azure in one place.